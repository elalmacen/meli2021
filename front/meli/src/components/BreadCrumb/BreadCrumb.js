import React from 'react';
import { Col } from 'react-bootstrap';
import Breadcrumb from 'react-bootstrap/Breadcrumb';

function BreadCrumb({categories}) {
    return (
        <Col className="breadcrumb-view">
            <Breadcrumb>
            {
                categories && categories.map((category, index) => <Breadcrumb.Item key={index} active={index === categories.length -1 ? true : false}>{category}</Breadcrumb.Item>)
            }
            </Breadcrumb>
        </Col>
    );
}

export default BreadCrumb;
