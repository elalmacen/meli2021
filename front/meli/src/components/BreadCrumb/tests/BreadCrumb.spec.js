import React from 'react';
import {shallow} from 'enzyme';
import BreadCrumb from '../BreadCrumb';


describe('test breadCrumb component', () => {
    test('renders without crashing', () => {
        const wrapper = shallow(<BreadCrumb />);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const categories = [
            'batman',
            'robin'
        ];
        const wrapper = shallow(<BreadCrumb categories={categories}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
