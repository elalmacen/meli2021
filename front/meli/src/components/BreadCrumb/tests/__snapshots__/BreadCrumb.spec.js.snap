// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test breadCrumb component renders without crashing 1`] = `
<Col
  className="breadcrumb-view"
>
  <Breadcrumb
    label="breadcrumb"
    listProps={Object {}}
  />
</Col>
`;

exports[`test breadCrumb component renders without crashing 2`] = `
<Col
  className="breadcrumb-view"
>
  <Breadcrumb
    label="breadcrumb"
    listProps={Object {}}
  >
    <BreadcrumbItem
      active={false}
      key="0"
      linkProps={Object {}}
    >
      batman
    </BreadcrumbItem>
    <BreadcrumbItem
      active={true}
      key="1"
      linkProps={Object {}}
    >
      robin
    </BreadcrumbItem>
  </Breadcrumb>
</Col>
`;
